ToDoApp/ToDoApp/ToDoListViewModel.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  ToDoListViewModel.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-09.
+//
+
+import Foundation
+
+final class ToDoListViewModel: ObservableObject {
+    enum State {
+        case idle
+        case loading
+        case loaded([ToDoSection])
+        case failed(Error)
+    }
+    @Published var state: State = .idle
+    private let taskService: ToDoService = FeaturesToDo()
```

Comment:
there should be two inits like in the CreateTaskViewModel, otherwise we would not be able to inject different implementation to the view model